sklearn/utils/estimator_checks.py

```diff
@@ -2361,12 +2367,28 @@ def check_outliers_train(name, estimator_orig, readonly_memmap=True):
             decision = estimator.decision_function(X)
             check_outlier_corruption(num_outliers, expected_outliers, decision)
 
-        # raises error when contamination is a scalar and not in [0,1]
-        msg = r"contamination must be in \(0, 0.5]"
-        for contamination in [-0.5, 2.3]:
-            estimator.set_params(contamination=contamination)
-            with raises(ValueError, match=msg):
-                estimator.fit(X)
+
+def check_outlier_contamination(name, estimator_orig):
+    # Check that the contamination parameter is in (0.0, 0.5] when it is an
+    # interval constraint.
+
+    if not hasattr(estimator_orig, "_parameter_constraints"):
+        # Only estimator implementing parameter constraints will be checked
+        return
+
+    contamination_constraints = estimator_orig._parameter_constraints["contamination"]
```

Comment:
the line above should not let it happen.

Reply:
Actually, it could be due to some inheritance thingy. `BaseNeighbors` is probably defining the `_parameter_constraint` but does not define a `"contamination"` constraint.

Reply:
This should be no longer be an issue after https://github.com/scikit-learn/scikit-learn/pull/23772#pullrequestreview-1030199471